registry/storage/driver/walk.go

```diff
@@ -22,9 +22,14 @@ type WalkFn func(fileInfo FileInfo) error
 // to a directory, the directory will not be entered and Walk
 // will continue the traversal.  If fileInfo refers to a normal file, processing stops
 func WalkFallback(ctx context.Context, driver StorageDriver, from string, f WalkFn) error {
+	_, err := doWalkFallback(ctx, driver, from, f)
+	return err
+}
+
+func doWalkFallback(ctx context.Context, driver StorageDriver, from string, f WalkFn) (bool, error) {
```

Comment:
For other reviewers, the explanation for this change can be found [here](https://github.com/digitalocean/docker-distribution/pull/17/files#r659974120).

Reply:
I included an explanation for this change in this PR description